lib/kino/process.ex

```diff
@@ -137,6 +139,205 @@ defmodule Kino.Process do
     """)
   end
 
+  @doc """
+  Generate a sequence diagram of process messages.
+
+  The provided function is executed and traced, with all the events sent to and
+  received by the trace target processes rendered in a sequence diagram. The trace
+  target argument can either be a single PID, a list of PIDs, or the atom `:all`
+  depending on what messages you would like to retain in your trace.
+
+  ## Examples
+
+  To generate a trace of all the messages occurring during the execution of the
+  provided function, you can do the following:
+
+      Kino.Process.seq_trace(:all, fn ->
+        {:ok, agent_pid} = Agent.start_link(fn -> [] end)
+        Process.monitor(agent_pid)
+
+        1..2
+        |> Task.async_stream(fn thing ->
+          Agent.get(agent_pid, fn thing -> thing end)
+          100 * thing
+        end,
+        max_concurrency: 3)
+        |> Stream.run()
+
+        Agent.stop(agent_pid)
+      end)
+
+  If you are only interested in messages being sent to or received by certain PIDs,
+  you can filter the sequence diagram by specifying the PIDs that you are interested
+  in:
+
+      {:ok, agent_pid} = Agent.start_link(fn -> [] end)
+      Process.monitor(agent_pid)
+
+      Kino.Process.seq_trace(agent_pid, fn ->
+        1..2
+        |> Task.async_stream(fn thing ->
+          Agent.get(agent_pid, fn thing -> thing end)
+          100 * thing
+        end,
+        max_concurrency: 3)
+        |> Stream.run()
+
+        Agent.stop(agent_pid)
+      end)
+  """
+  @spec seq_trace(trace_target(), (() -> any())) :: Markdown.t()
+  def seq_trace(pid, trace_function) when is_pid(pid) do
+    seq_trace([pid], trace_function)
+  end
+
+  def seq_trace(trace_pids, trace_function) do
+    # Set up the process message tracer and the Erlang seq_trace_module
+    {:ok, tracer_pid} = Tracer.start_link(nil)
+    :seq_trace.set_token(:send, true)
+    :seq_trace.set_token(:receive, true)
+    :seq_trace.set_token(:monotonic_timestamp, true)
+    previous_tracer = :seq_trace.set_system_tracer(tracer_pid)
+
+    # Run the user supplied function
+    trace_function.()
+
+    # Reset all of the tracing options
+    :erlang.trace(:all, false, [:send, :receive])
```

Comment:
I think we don't need to change Erlang trace, do we?

Reply:
Ooops. That was copy and paste carry over from when I was experimenting with trace versus seq_trace. Fixed!